emacs/.emacs.d/settings.org
* Package Repos

#+begin_src emacs-lisp
  (require 'package)
  ;; Sets a list of package archive mirrors
  (setq package-archives '(("gnu" . "https://elpa.gnu.org/packages/")
                           ("melpa" . "https://melpa.org/packages/")
                           ("org" . "https://orgmode.org/elpa/")
                           ("elpa" . "https://elpa.gnu.org/packages/")))
  (package-initialize)

  ;; Keep packages up to date.
  (unless package-archive-contents
    (package-refresh-contents))

  ;; Use-package is a clean macro for package installing and configuration.
  (unless (package-installed-p 'use-package)
    (package-install 'use-package))

  (require 'use-package)
#+end_src

* UI Features

#+begin_src emacs-lisp
  ;; Automatically detect file updates
  (global-auto-revert-mode t)

  (scroll-bar-mode -1)
  (menu-bar-mode -1) 
  (tool-bar-mode -1)
  (column-number-mode 1)
  (delete-selection-mode)
  (setq-default truncate-lines 1)
  (setq-default word-wrap 1)
                                          ;(setq split-window-preferred-function 'split-window-horizontally)
 #+end_src

* Theme and Font

#+begin_src emacs-lisp
  (defconst *theme-directory* "~/.emacs.d/themes/")

  (add-to-list 'custom-theme-load-path *theme-directory*)
  (load-theme 'zenburn t)

  (set-frame-font "Jetbrains Mono-11" nil t)
  (global-font-lock-mode 1) ; ??
#+end_src

* Keybinds

#+begin_src emacs-lisp
  (use-package which-key
    :ensure t
    :config
    (setq which-key-show-early-on-C-h nil)
    (setq which-key-idle-delay 0.5)
    (setq which-key-idle-secondary-delay 0.05)
    (which-key-setup-minibuffer)
    (which-key-setup-side-window-bottom)
    (setq which-key-popup-type 'minibuffer)
    (which-key-mode))
#+end_src

* Org Mode
#+begin_src emacs-lisp
    (setq org-agenda-file '("~/org"))
    (setq org-hide-emphasis-markers t)
    (setq org-return-follows-link t)

    ;; Shortcuts for storing links, viewing the agenda, and starting a capture
    (define-key global-map "\C-cl" 'org-store-link)
    (define-key global-map "\C-ca" 'org-agenda)
    (define-key global-map "\C-cc" 'org-capture)

    ;; Adjust height of headings
    (let* ((variable-tuple
          (cond ((x-list-fonts "ETBembo")         '(:font "ETBembo"))
                ((x-list-fonts "Source Sans Pro") '(:font "Source Sans Pro"))
                ((x-list-fonts "Lucida Grande")   '(:font "Lucida Grande"))
                ((x-list-fonts "Verdana")         '(:font "Verdana"))
                ((x-family-fonts "Sans Serif")    '(:family "Sans Serif"))
                (nil (warn "Cannot find a Sans Serif Font.  Install Source Sans Pro."))))
         (base-font-color     (face-foreground 'default nil 'default))
         (headline           `(:inherit default :weight bold :foreground ,base-font-color)))

    (custom-theme-set-faces
     'user
     `(org-level-8 ((t (,@headline ,@variable-tuple))))
     `(org-level-7 ((t (,@headline ,@variable-tuple))))
     `(org-level-6 ((t (,@headline ,@variable-tuple))))
     `(org-level-5 ((t (,@headline ,@variable-tuple))))
     `(org-level-4 ((t (,@headline ,@variable-tuple :height 1.1))))
     `(org-level-3 ((t (,@headline ,@variable-tuple :height 1.2))))
     `(org-level-2 ((t (,@headline ,@variable-tuple :height 1.3))))
     `(org-level-1 ((t (,@headline ,@variable-tuple :height 1.5))))
     `(org-document-title ((t (,@headline ,@variable-tuple :height 1.6 :underline nil))))))

    ;; Load languages to execute in code blocks
  
  ;(org-babel-do-load-languages
  ; 'org-babel-load-languages
  ; '((emacs-lisp . t)
  ;   (racket . t)))


    (add-hook 'org-mode-hook 'org-indent-mode)
    (add-hook 'org-mode-hook 'visual-line-mode)
#+end_src 

* Programming Modes
#+begin_src emacs-lisp
  (use-package flycheck
    :ensure t)

  (use-package flymake
    :ensure t)

  (use-package company
    :ensure t)

    (use-package eglot
    :ensure t)
#+end_src

** C
#+begin_src emacs-lisp
  (add-hook 'c-mode-hook (lambda () (c-set-style "k&r")))
#+end_src

** GNU APL
#+begin_src emacs-lisp
  (use-package gnu-apl-mode
    :ensure t)
#+end_src

** Racket
#+begin_src emacs-lisp
  (use-package racket-mode
    :ensure t
    :config
    (add-hook 'racket-mode-hook #'eglot))
#+end_src
** Go
#+begin_src emacs-lisp
  (use-package go-mode
    :ensure t
    :config
    );(add-hook 'go-mode-hook #'eglot))
#+end_src
* PDF Viewing/Editing
#+begin_src emacs-lisp
  (use-package pdf-tools
    :ensure t
    :config
    (pdf-tools-install))
#+end_src
* LaTeX
#+begin_src emacs-lisp
  (use-package auctex
    :ensure t
    :config
    (setq TeX-auto-save t)
    (setq TeX-parse-self t)
    (setq-default TeX-master nil)
    (prettify-symbols-mode))
#+end_src



emacs/.emacs.d/settings.el
(require 'package)
;; Sets a list of package archive mirrors
(setq package-archives '(("gnu" . "https://elpa.gnu.org/packages/")
			 ("melpa" . "https://melpa.org/packages/")
			 ("org" . "https://orgmode.org/elpa/")
			 ("elpa" . "https://elpa.gnu.org/packages/")))
(package-initialize)

;; Keep packages up to date.
(unless package-archive-contents
  (package-refresh-contents))

;; Use-package is a clean macro for package installing and configuration.
(unless (package-installed-p 'use-package)
  (package-install 'use-package))

(require 'use-package)

(scroll-bar-mode -1)
(menu-bar-mode -1) 
(tool-bar-mode -1)
(column-number-mode 1)
(delete-selection-mode)
(setq-default truncate-lines 1)
(setq-default word-wrap 1)
;(setq split-window-preferred-function 'split-window-horizontally)

;; C indenting
(setq c-default-style "linux"
      c-basic-offset 4)

(defconst *mood-line* "~/.emacs.d/mood-line.el")
(load-file *mood-line*)
(mood-line-mode)

(defconst *theme-directory* "~/.emacs.d/themes/")

(add-to-list 'custom-theme-load-path *theme-directory*)
(load-theme 'zenburn t)
(set-frame-font "Fantasque Sans Mono 16" nil t)

(use-package selectrum
  :ensure t
  :config (selectrum-mode 1))

(use-package selectrum-prescient
  :ensure t)
(selectrum-prescient-mode 1)
(prescient-persist-mode 1)

(use-package which-key
  :ensure t)
(setq which-key-show-early-on-C-h 1)
(setq which-key-idle-delay 0.5)
(setq which-key-idle-secondary-delay 0.05)
(which-key-setup-minibuffer)
(which-key-setup-side-window-bottom)
(setq which-key-popup-type 'minibuffer)
(which-key-mode)

(setq inhibit-startup-message t)

(add-hook 'org-mode-hook
		(lambda () (visual-line-mode)))

     (setq org-agenda-files (directory-files-recursively "~/org/" "\\.org$"))

; Targets include this file and any file contributing to the agenda - up to 9 levels deep
(setq org-refile-targets (quote ((nil :maxlevel . 9))))
     
     (setq org-support-shift-select t)
     (setq org-ellipsis " ▾")
     (set-face-underline 'org-ellipsis nil)

(use-package lsp-mode
  :ensure t
  :config (setq lsp-keymap-prefix "C-l"))

(use-package lsp-ui
  :ensure t)

(use-package flycheck
  :ensure t)

(use-package company
  :ensure t
  :config (setq company-idle-delay 0.01))

(add-hook 'prog-mode-hook
	  (lambda () (progn (display-line-numbers-mode t)
			    (hs-minor-mode t)
			    (setq display-line-numbers-type 'relative))))

(use-package slime :ensure)
(use-package slime-company :ensure)
(setq inferior-lisp-program "/usr/local/bin/sbcl --noinform")
(slime-setup '(slime-fancy slime-company))

(add-hook 'c-mode-hook
	  (lambda () (progn (lsp-deferred)
			    (flycheck-mode t)
			    (company-mode t))))

(add-hook 'c++-mode-hook
	  (lambda () (progn (lsp-deferred)
			    (flycheck-mode t)
			    (company-mode t))))
